components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/AbstractJWTGenerator.java

```diff
@@ -377,4 +384,52 @@ protected Application getApplicationById(String tenantDomain, int applicationId)
         SubscriptionDataStore datastore = SubscriptionDataHolder.getInstance().getTenantSubscriptionStore(tenantDomain);
         return datastore.getApplicationById(applicationId);
     }
+
+    /**
+     * Utility method to generate JWT header with public certificate thumbprint for signature verification.
+     *
+     * @param publicCert         The public certificate which needs to include in the header as thumbprint
+     * @param signatureAlgorithm Signature algorithm which needs to include in the header
+     * @param useKid             Boolean to indicate whether to include kid property in the header
+     */
+    public static String generateHeader(Certificate publicCert, String signatureAlgorithm, boolean useKid)
+            throws APIManagementException {
+        try {
+            //generate the SHA-1 thumbprint of the certificate
+            MessageDigest digestValue = MessageDigest.getInstance("SHA-1");
+            byte[] der = publicCert.getEncoded();
+            digestValue.update(der);
+            byte[] digestInBytes = digestValue.digest();
+            String publicCertThumbprint = APIUtil.hexify(digestInBytes);
+            String base64UrlEncodedThumbPrint;
+            base64UrlEncodedThumbPrint = java.util.Base64.getUrlEncoder()
+                    .encodeToString(publicCertThumbprint.getBytes(StandardCharsets.UTF_8));
+            java.security.cert.X509Certificate x509Certificate = (java.security.cert.X509Certificate) publicCert;
+            StringBuilder jwtHeader = new StringBuilder();
+            /*
+             * Sample header
+             * {"typ":"JWT", "alg":"SHA256withRSA", "x5t":"a_jhNus21KVuoFx65LmkW2O_l10",
+             * "kid":"a_jhNus21KVuoFx65LmkW2O_l10_RS256"}
+             * {"typ":"JWT", "alg":"[2]", "x5t":"[1]", "x5t":"[1]"}
+             * */
+            jwtHeader.append("{\"typ\":\"JWT\",");
+            jwtHeader.append("\"alg\":\"");
+            jwtHeader.append(APIUtil.getJWSCompliantAlgorithmCode(signatureAlgorithm));
+            jwtHeader.append("\",");
+
+            jwtHeader.append("\"x5t\":\"");
+            jwtHeader.append(base64UrlEncodedThumbPrint);
+            jwtHeader.append("\"");
+            if (useKid) {
+                jwtHeader.append(",\"kid\":\"");
+                jwtHeader.append(JWTUtil.getKID(x509Certificate));
+                jwtHeader.append("\"");
+            }
+            jwtHeader.append("}");
+            return jwtHeader.toString();
+
+        } catch (Exception e) {
```

Comment:
Bad practice to catch generic exception

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12091/commits/5ca931825782d3665e7e34ad7cc3da9ee185ef8f